.github/workflows/automatic_code_review.yml

```diff
@@ -0,0 +1,59 @@
+name: Automatic code review
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  review_code:
+    name: Run Danger 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Checkout source code
+        uses: actions/checkout@v2.3.2
+
+      - name: Load Gradle cached
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      # Run Detekt and Archive Detekt reports
+      - name: Run Detekt on CoroutineTemplate
+        working-directory: ./CoroutineTemplate
+        run: ./gradlew detekt
+
+      - name: Run Detekt on RxJavaTemplate
+        working-directory: ./RxJavaTemplate      
+        run: ./gradlew detekt
+
+      - name: Archive Detekt report on CoroutineTemplate
+        uses: actions/upload-artifact@v2
+        with:
+          name: detekt-report-coroutine
+          path: CoroutineTemplate/build/reports/detekt/
+
+      - name: Archive Detekt report on RxJavaTemplate
+        uses: actions/upload-artifact@v2
+        with:
+          name: detekt-report-rxjava
+          path: RxJavaTemplate/build/reports/detekt/
+
+      - name: Run Android Lint on RxJavaTemplate
+        working-directory: ./RxJavaTemplate      
+        run: ./gradlew lint
+
+      - name: Danger
+        uses: danger/kotlin@1.0.0-beta.3
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't feel like why wouldn't we just add this config to the current workflow that we already verify the `detekt` before. The more workflow we had and duplicating the same steps, the longer it takes and costs from the CI service.
Is there any specific reason for a new separated workflow file, @minhnimble ? 🤔

Reply:
and eof linebreak

Reply:
@sleepylee for the creating a new workflow part, I was just trying keep a similar fashion with the Web Chapter implementation whereas they created a separate workflow for Danger. More reference

Reply:
> At the same time, the triggers are different since there are some checks in Danger that can only be executed with pull requests related events, not push events.
> 
> Thus, it is really hard to reuse the current workflow. If you have any better suggestion please let me know 🙏

@minhnimble I'm wondering can we apply this condition for the steps related to `danger`: `if: ${{ github.event.issue.pull_request }}` in the current workflow. I feel repeating all steps to trigger `danger` might not be a good approach 😅

Reply:
@manh-t it is not working and the job is completely skipped even though I triggered with both push and pull_request events 😅

<img width="915" alt="Screen Shot 2021-09-28 at 16 33 46" src="https://user-images.githubusercontent.com/70877098/135062519-b473c6ae-bba9-4871-85d3-56b6b20ec211.png">

Here is the code I applied:

<img width="402" alt="Screen Shot 2021-09-28 at 16 34 04" src="https://user-images.githubusercontent.com/70877098/135062599-8b94ae95-6ceb-499e-bc6a-72b36a1a187f.png">

Reply:
@minhnimble my bad, it's the `if` condition for `workflow` 😂 in `job`, we may write this `if: ${{ github.event_name == 'pull_request' }}`, [reference](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idif) 🙏

Reply:
@manh-t `if: ${{ github.event_name == 'pull_request' }}` can't work with `push` trigger. We will have to update the trigger of our current workflow to `pull_request` instead of `push`, but we don't want that, right? 😅

<img width="828" alt="Screen Shot 2021-09-28 at 17 04 05" src="https://user-images.githubusercontent.com/70877098/135068067-d894dc64-0245-47db-bb91-baeb22b8fe3b.png">

Reply:
@minhnimble alright, there are some edge cases. We can improve it later, I think 😥